### PR TITLE
Rishabh7m patch 6

### PR DESCRIPTION
**Pull request by rishabh7m**
gst_player_set_playback_rate() uses gst_player_seek()

### CODE REVIEW

**Comment by sdroege** [comment]
Please squash all the commits together into one

File: lib/gst/player/gstplayer.c [hunk 1]
```diff
@@ -1443,10 +1450,18 @@ gst_player_seek_internal_locked (GstPlayer * self)
 
   remove_tick_source (self);
   self->priv->is_eos = FALSE;
+  
+  start_pos = gst_player_get_position(self);
```
Comment (sdroege) [inline lib/gst/player/gstplayer.c]: This now breaks seeks, you would now always seek to the current position. The seek function needs some refactoring to handle both cases.

File: lib/gst/player/gstplayer.c [hunk 2]
```diff
@@ -688,6 +688,13 @@ eos_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
   self->priv->is_eos = TRUE;
 }
 
+void gst_player_set_playback_rate (GstPlayer * self, gdouble rate)
+{
+  g_return_if_fail (GST_IS_PLAYER (self));
+  self->priv->rate = rate;
+  gst_player_seek (self, 0);
```
Comment (ened) [inline lib/gst/player/gstplayer.c]: Is this for playback speed? If so, shouldn't it continue at the same position?

File: lib/gst/player/gstplayer.c [hunk 3]
```diff
@@ -688,6 +688,13 @@ eos_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
   self->priv->is_eos = TRUE;
 }
 
+void gst_player_set_playback_rate (GstPlayer * self, gdouble rate)
```
Comment (ened) [inline lib/gst/player/gstplayer.c]: Should we add this to `gstplayer.h`?

File: lib/gst/player/gstplayer.c (hunk 2 above)
Comment (sdroege) [inline lib/gst/player/gstplayer.c]: Yes, see my comments. This all has to be done differently and refactored to have unified seeking logic for play speed and seeking.

**Comment by sdroege** [comment]
This was now solved by another pull request

**Pull request closed without merging**